sdk/src/Core/Amazon.Runtime/Internal/Settings/PersistenceManager.cs

```diff
@@ -253,10 +254,18 @@ SettingsCollection loadSettingsType(string type)
                         content = reader.ReadToEnd();
                     }
 
-                    var settings = JsonMapper.ToObject<Dictionary<string, Dictionary<string, object>>>(content);
+                    Dictionary<string, Dictionary<string, string>> settings;
+                    if(!string.IsNullOrEmpty(content))
+                    {
+                        settings = JsonSerializerHelper.Deserialize<Dictionary<string, Dictionary<string, string>>>(content, DictionaryStringDictionaryStringJsonSerializerContexts.Default);
 
-                    if (settings == null)
-                        settings = new Dictionary<string, Dictionary<string, object>>();
+                        if (settings == null)
+                            settings = new Dictionary<string, Dictionary<string, string>>();
```

Comment:
nit: this could be done outside the if and scratch the else completely to remove the duplicate code.